### PR TITLE
Chore: e2e tests on text input component (Issue/138)

### PR DESCRIPTION
**Pull request by lemmyadams**
Addresses #138

### CODE REVIEW

**Comment by lemmyadams** [comment]
Relies on https://github.com/adaptlearning/adapt_framework/pull/3533

File: test/e2e/textInput.cy.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    cy.visit('/');
+  });
+
+  it('should display the text input component', function () {
+    const textInputComponents = this.data.components.filter((component) => component._component === 'textinput')
+    textInputComponents.forEach((textInputComponent) => {
+      cy.visit(`/#/preview/${textInputComponent._id}`);
+      cy.stripHtml(textInputComponent.body)
+      const bodyWithoutHtml = this.text;
+      cy.testContainsOrNotExists('.textinput__body', bodyWithoutHtml)
```
Comment (oliverfoster) [inline test/e2e/textInput.cy.js:13]: Where the body is `"this is <b>bold</b> text"`, the stripped string would be `"this is bold text"` which will not going to return true here `cy.get('.textinput__body').should('contain', "this is bold text");` I think.

"Get the DOM element containing the text. DOM elements can contain more than the desired text and still match"

Reference:
* (cypress.contains)[https://docs.cypress.io/api/commands/contains]

Comment (cahirodoherty-learningpool) [reply test/e2e/textInput.cy.js:13]: Hi Oliver, just been discussing this with Lemmy and we think that this achieves what we want from our tests.

Example 1:
- displayTitle set to `"this is <b>bold</b> text"`
- test against stripped text `"this is bold text"`
- test passes as expected ✅ 

Example 2:
- displayTitle set to `"this is not bold text"`
- test against stripped text `"this is not bold text"`
- test passes as expected ✅ 

Example 3:
- displayTitle set to `"this is other text </br>"`
- test against stripped text `"this is other text"`
- test passes as expected ✅ 

Where "test" in the above refers to `cy.get('.textinput__body').should('contain', strippedText);`

Comment (lemmyadams) [reply test/e2e/textInput.cy.js:13]: As addition to above, yes "Get the DOM element containing the text. DOM elements can contain more than the desired text and still match" is correct, but that's still assuming the exact text is contained within a larger text. 
Without stripping the html, the test checks if it contains the variable supplied with the component, which in raw form can contain html and will be converted on the page, is what exactly is rendered on the page, without the html being converted. Thus you get a bold text, which will appear as '<b>bold</b>' in the var on the test, but only 'bold' on the page, and so the test fails.

Comment (oliverfoster) [reply test/e2e/textInput.cy.js:13]: Yes, it wasn't clear to me that cypress contains was doing overall element text. Thanks for verifying.

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]

File: test/e2e/textInput.cy.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  it('should display the text input component', function () {
+    const textInputComponents = this.data.components.filter((component) => component._component === 'textinput')
+    textInputComponents.forEach((textInputComponent) => {
+      cy.visit(`/#/preview/${textInputComponent._id}`);
+      const bodyWithoutHtml = cy.helpers.stripHtml(textInputComponent.body)
+      cy.testContainsOrNotExists('.textinput__body', bodyWithoutHtml)
+
+      cy.testContainsOrNotExists('.textinput__title', textInputComponent.displayTitle)
+      cy.testContainsOrNotExists('.textinput__instruction', textInputComponent.instruction)
```
Comment (oliverfoster) [inline test/e2e/textInput.cy.js:15]: Both of these properties can contain html as well.

File: test/e2e/textInput.cy.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    cy.visit('/');
+  });
+
+  it('should display the text input component', function () {
+    const textInputComponents = this.data.components.filter((component) => component._component === 'textinput')
+    textInputComponents.forEach((textInputComponent) => {
+      cy.visit(`/#/preview/${textInputComponent._id}`);
+      const bodyWithoutHtml = cy.helpers.stripHtml(textInputComponent.body)
+      cy.testContainsOrNotExists('.textinput__body', bodyWithoutHtml)
+
```
Comment (oliverfoster) [inline test/e2e/textInput.cy.js:13]: Odd whitespace

File: test/e2e/textInput.cy.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+describe('Text Input', function () {
+  beforeEach(function () {
+    cy.getData()
+    cy.visit('/');
+  });
+
+  it('should display the text input component', function () {
+    const textInputComponents = this.data.components.filter((component) => component._component === 'textinput')
+    textInputComponents.forEach((textInputComponent) => {
```
Comment (oliverfoster) [inline test/e2e/textInput.cy.js:9]:
```suggestion
    textInputComponents.forEach(textInputComponent => {
```

These brackets aren't needed, please remove them from fat-arrow functions.

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#syntax

**Review by oliverfoster — CHANGES_REQUESTED** [review]

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**